spec/acceptance/graphite_exporter_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'spec_helper_acceptance'
+
+describe 'prometheus graphite exporter' do
+  it 'graphite_exporter works idempotently with no errors' do
+    pp = 'include prometheus::graphite_exporter'
+    apply_manifest(pp, catch_failures: true)
+    apply_manifest(pp, catch_changes: true)
+  end
+
+  describe service('graphite_exporter') do
+    it { is_expected.to be_running }
+    it { is_expected.to be_enabled }
+  end
+  describe port(9109) do
+    it { is_expected.to be_listening.with('tcp6') }
+  end
+
+  describe 'graphite_exporter update from 0.2.0 to 0.6.2' do
+    it 'is idempotent' do
+      pp = "class{'prometheus::graphite_exporter': version => '0.2.0'}"
+      apply_manifest(pp, catch_failures: true)
+      apply_manifest(pp, catch_changes: true)
+    end
+
+    describe service('graphite_exporter') do
+      it { is_expected.to be_running }
+      it { is_expected.to be_enabled }
+    end
+
+    describe port(9109) do
+      it { is_expected.to be_listening.with('tcp6') }
+    end
+    it 'is idempotent' do
+      pp = "class{'prometheus::graphite_exporter': version => '0.6.2'}"
```

Comment:
If you are going to specify the version, you should check that the specified version is what got installed.

Reply:
I am not the author and this branch does not belong to my namespace.